### PR TITLE
MGMT-7569: remove release getters from Versions

### PR DESCRIPTION
**Pull request by danielerez**
# Assisted Pull Request

## Description

In order to simplify the Versions API and avoid code duplication, removed the GetReleaseImage and GetReleaseVersion functions. Instead, a single func (GetOpenshiftVersion) that returns the entire OpenshiftVersion struct should be used.
That way, extending the struct with additional fields should be easier, as the validation code is encapsulated in one place.

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

Just an internal code-change.

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

Added relevant unit-tests.

## Assignees

/cc @YuviGold 
/cc @filanov 
/cc @osherdp 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -1740,14 +1740,14 @@ func (b *bareMetalInventory) generateClusterInstallConfig(ctx context.Context, c
...
 
-	releaseImage, err := b.versionsHandler.GetReleaseImage(cluster.OpenshiftVersion)
-
+	ocpVersion, err := b.versionsHandler.GetOpenshiftVersion(cluster.OpenshiftVersion)
 	if err != nil {
-		log.WithError(err).Errorf("failed to get release image for cluster %s with openshift version %s", cluster.ID, cluster.OpenshiftVersion)
-		return errors.Wrapf(err, "failed to get release image for cluster %s with openshift version %s", cluster.ID, cluster.OpenshiftVersion)
+		msg := fmt.Sprintf("failed to get OpenshiftVersion for cluster %s with openshift version %s", cluster.ID, cluster.OpenshiftVersion)
+		log.WithError(err).Errorf(msg)
+		return errors.Wrapf(err, msg)
```
Comment (osherdp) [inline internal/bminventory/inventory.go:1747]: why log the error and then return it upwards?
seems like something that only leads to printing the same error on two places
iiuc just returning it with the wrapping is enough

**Review by osherdp — CHANGES_REQUESTED** [review]

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (danielerez) [reply internal/bminventory/inventory.go:1747]: We actually do the same for the other errors in generateClusterInstallConfig, need to check if redundant.

Comment (danielerez) [reply internal/bminventory/inventory.go:1747]: Verified that the log is indeed printed only once. So the log seems mandatory.

**Review by osherdp — APPROVED** [review]

**Pull request merged**